### PR TITLE
Add :unique_index option to #create_translation_table!

### PR DESCRIPTION
**Pull request by ridiculous**
What do you think about supporting a new `:unique_index` option that can be passed to the `Model.create_translation_table!` method to create a new unique index on `locale` and `model_id`?

### CODE REVIEW

**Comment by ridiculous** [comment]
Looks like Travis is failing on an unrelated Rails 5 thing..

**Comment by parndt** [comment]
I think it would be best if this is handled using a manually created database migration, so that we can avoid the extra complexity.  Is that okay, @ridiculous ?

**Comment by ridiculous** [comment]
Yeah, that's what we've been doing. It's a hassle though, cause we gotta drop the index added by Globalize and then re-add the same one, but with the unique constraint. Is really that more complex?

**Comment by ridiculous** [comment]
Even if we don't support the new `:unique_index` option I'm proposing, I think we should still validate the given options, for the purpose of being more user friendly

**Comment by parndt** [comment]
@ridiculous ah I didn't understand you had to drop and re-create it.. gotcha. Then this might be worthwhile after all!

File: lib/globalize/active_record/migration.rb [hunk 1]
```diff
@@ -178,6 +191,14 @@ def clear_schema_cache!
 
         private
 
+        def truncate_index_name(index_name)
```
Comment (parndt) [inline lib/globalize/active_record/migration.rb]: 👍

**Pull request merged**